compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1287,17 +1287,21 @@ pub fn ice_path() -> &'static Option<PathBuf> {
         if !rustc_feature::UnstableFeatures::from_environment(None).is_nightly_build() {
             return None;
         }
-        if let Ok("0") = std::env::var("RUST_BACKTRACE").as_deref() {
+        if let Some(s) = std::env::var_os("RUST_BACKTRACE") && s == "0" {
```

Comment:
There should be no difference here. `env::var` will return `Err` in case of a non-UTF-8 value and `env::var_os` will return a value not comparing equal to `"1"`.

Reply:
I agree, but this avoids the appearance of an unhandled error.

Would you prefer I revert this?

Reply:
Yes, please revert this and elsewhere.

Reply:
Actually, no - this is different; the old code would treat non-UTF8 paths as invalid and call env::current_dir. This now correctly handles non-unicode paths.

Reply:
The place where a path is passed in an env var can remain the same, but the rest can be reverted without behavior changes.

Reply:
You're right - I went ahead and change all interactions with `RUST_BACKTRACE` to `env_os` for consistency and updated the commit message.